src/main/java/org/springframework/integration/dsl/context/IntegrationFlowContext.java

```diff
@@ -147,12 +144,9 @@ public void register(String flowId, IntegrationFlow integrationFlow, boolean aut
 	 */
 	public synchronized void remove(String flowId) {
 		if (this.registry.containsKey(flowId)) {
-			IntegrationFlow theFlow = this.registry.remove(flowId);
-			if (theFlow instanceof Lifecycle) {
-				((Lifecycle) theFlow).stop();
-			}
+			IntegrationFlowRegistration flowRegistration = this.registry.remove(flowId);
+			flowRegistration.stop();
 			((DefaultSingletonBeanRegistry) this.beanFactory).destroySingleton(flowId);
-			this.flowInputChannelCache.remove(flowId);
```

Comment:
No need. See `registerDependentBean()` in the `registerBean()` here and an additional `assertFalse(this.beanFactory.containsBean()` in the test-case.